tests/test_examples.py

```diff
@@ -1,32 +1,26 @@
-import pytest
-
-
-def test_decaying_moving_eddy():
-    from parcels.examples.example_decaying_moving_eddy import main
-    main()
-
-
-def test_moving_eddies():
-    from parcels.examples.example_moving_eddies import main
-    main(['jit'])
-
-
-@pytest.mark.xfail(reason="example data not available from within unit-tests")
-def test_nemo_curvilinear():
-    from parcels.examples.example_nemo_curvilinear import main
-    main(['jit'])
+import runpy
+import sys
+from pathlib import Path
 
+import pytest
 
-def test_peninsula():
-    from parcels.examples.example_peninsula import main
-    main(['jit'])
+# TODO: Broaden scope to all `.py` scripts in `example_folder`?
+example_folder = (Path(__file__).parent / "../docs/examples").resolve()
+example_fnames = [
+    "example_decaying_moving_eddy.py",
+    "example_moving_eddies.py",
+    # "example_nemo_curvilinear.py", #! Imports cartopy, which is not installed in CI
+    "example_peninsula.py",
+    "example_radial_rotation.py",
+    "example_stommel.py",
```

Comment:
Done 7bafe84db3f8caa4dc8cb16dfcc1974fee543efb. Updated script finding to eager finding and testing `examples/*.py` (rather than hardcoded files). An FYI, most of these extra examples don't define a `main()` function, so the test case isn't really useful here (although they do define pytests).